ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -292,4 +338,66 @@ protected String getReplicaThreadName(String datacenterToReplicateFrom, int thre
   protected boolean shouldReplicateFromDc(String datacenterName) {
     return cloudConfig.vcrSourceDatacenters.contains(datacenterName);
   }
+
+  /**
+   * Check and schedule a VCR helix update task. We schedule updateVcrHelix in 2 cases:
+   * 1. On a node become online role of partition 1
+   * 2. On Ambry cluster change
+   */
+  private void scheduleVcrHelix() {
+    if (vcrHelixUpdateFuture != null && vcrHelixUpdateFuture.cancel(false)) {
+      // If a vcrHelixUpdate task is scheduled, try to cancel it first.
+      logger.info("There was a scheduled vcrHelixUpdate task. Canceled.");
+    }
+    // either success cancel or not, we should schedule a new job to updateVcrHelix
+    vcrHelixUpdateFuture =
+        scheduler.schedule(() -> updateVcrHelix(), cloudConfig.vcrHelixUpdateDelayTimeInSeconds, TimeUnit.SECONDS);
+    logger.info("VcrHelixUpdate task scheduled. Will run in {} seconds.", cloudConfig.vcrHelixUpdateDelayTimeInSeconds);
+  }
+
+  /**
+   * The actual performer to update VCR Helix:
+   */
+  private void updateVcrHelix() {
+    String localDcZkStr = ((HelixClusterManager) clusterMap).getLocalDcZkConnectString();
+    logger.info("Going to update VCR Helix Cluster. Dryrun: {}", cloudConfig.vcrHelixUpdateDryRun);
+    logger.info("Current partitions in clustermap: ", clusterMap.getAllPartitionIds(null));
+    for (PartitionId partitionIdx : clusterMap.getAllPartitionIds(null)) {
+      logger.info(partitionIdx.toString());
+    }
+    try {
+      isVcrHelixUpdateInProgress = true;
+      HelixVcrUtil.updateResourceAndPartition(localDcZkStr, clusterMapConfig.clusterMapClusterName,
```

Comment:
Is it possible that this call fails (maybe temporary network issue)? In this case do we wait till the next time cluster map update happens on the main cluster?

Reply:
In this case, it highly possible that some fault error happened. My plan is to fire an alert and developers to look at it.